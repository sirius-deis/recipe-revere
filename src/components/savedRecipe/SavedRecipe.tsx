import { FC } from "react";
import { Link } from "react-router-dom";
import styles from "./SavedRecipe.module.css";

interface SavedRecipeProps {
  imgUrl: string;
  title: string;
  url: string;
}

const SavedRecipe: FC<SavedRecipeProps> = ({ imgUrl, title, url }) => {
  return (
    <article className={styles.container}>
      <div className={styles["image-container"]}>
        <img src={imgUrl} alt={title} className={styles.img} />
      </div>
      <h3 className={styles.title}>{title}</h3>
      <Link to={url} target="_blank" rel="noreferrer" className={styles.link}>
        Details
      </Link>
    </article>
  );
};

export default SavedRecipe;
