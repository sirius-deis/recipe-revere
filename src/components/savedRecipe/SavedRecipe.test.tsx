import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import SavedRecipe from "./SavedRecipe";

describe("SavedRecipe component", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <BrowserRouter>
        <SavedRecipe
          imgUrl="https://spoonacular.com/recipeImages/623856-556x370.jpg"
          title="Spaghetti Bolognese"
          url="https://spoonacular.com/recipeImages/623856-556x370.jpg"
        />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
  it("should render passed props correctly", () => {
    render(
      <BrowserRouter>
        <SavedRecipe
          imgUrl="https://spoonacular.com/recipeImages/623856-556x370.jpg"
          title="Spaghetti Bolognese"
          url="https://spoonacular.com/recipeImages/623856-556x370.jpg"
        />
      </BrowserRouter>
    );
    expect(screen.getByText("Spaghetti Bolognese")).toBeInTheDocument();
    const img = screen.getByRole("img");
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute(
      "src",
      "https://spoonacular.com/recipeImages/623856-556x370.jpg"
    );
    expect(img).toHaveAttribute("alt", "Spaghetti Bolognese");
    const link = screen.getByRole("link");
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute(
      "href",
      "https://spoonacular.com/recipeImages/623856-556x370.jpg"
    );
  });
});
