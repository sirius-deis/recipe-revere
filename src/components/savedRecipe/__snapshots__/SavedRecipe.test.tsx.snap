// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedRecipe component should match snapshot 1`] = `
<div>
  <article
    class="container"
  >
    <div
      class="image-container"
    >
      <img
        alt="Spaghetti Bolognese"
        class="img"
        src="https://spoonacular.com/recipeImages/623856-556x370.jpg"
      />
    </div>
    <h3
      class="title"
    >
      Spaghetti Bolognese
    </h3>
    <a
      class="link"
      href="https://spoonacular.com/recipeImages/623856-556x370.jpg"
      rel="noreferrer"
      target="_blank"
    >
      Details
    </a>
  </article>
</div>
`;
